File: backend/app/schemas/notification.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class NotificationResponse(BaseModel):
    id: str
    type: str
    title: str
    message: str
    is_read: bool
    channel: str
    sent_at: Optional[datetime] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationPreferenceResponse(BaseModel):
    email_notifications: bool = True
    push_notifications: bool = True
    weekly_report: bool = True
    streak_reminder: bool = True

    model_config = {"from_attributes": True}


class NotificationPreferenceUpdate(BaseModel):
    email_notifications: Optional[bool] = None
    push_notifications: Optional[bool] = None
    weekly_report: Optional[bool] = None
    streak_reminder: Optional[bool] = None

File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  user_id: string;
  type: "reminder" | "streak_warning" | "report_ready" | "system";
  title: string;
  message: string;
  is_read: boolean;
  channel: "in_app" | "email" | "push";
  sent_at: string | null;
  created_at: string;
}

export interface NotificationPreference {
  id?: string;
  user_id?: string;
  timezone: string;
  email_notifications: boolean;
  push_notifications: boolean;
  daily_reminder: boolean;
  reminder_time: string;
  weekly_report_enabled: boolean;
}

export interface NotificationPreferenceUpdate {
  timezone?: string;
  email_notifications?: boolean;
  push_notifications?: boolean;
  daily_reminder?: boolean;
  reminder_time?: string;
  weekly_report_enabled?: boolean;
}
